atc/engine/check_delegate_test.go

```diff
@@ -319,6 +329,29 @@ var _ = Describe("CheckDelegate", func() {
 						Expect(run).To(BeTrue())
 					})
 				})
+
+				Context("when the last check gets updated after the first loop", func() {
```

Comment:
What is this intended to test? I was expecting it to test that if it can't acquire the lock initially, it'll keep re-checking the `LastCheck` periodically, but it's really just testing the same thing as `when the interval has not elapsed since the last check`

Reply:
Yaaaa right, it is exactly like that test. I don't know what I was thinking when I added this test haha 😆  

I think maybe I wanted to test that the last check value gets rechecked everytime it loops (but ofc I didn't write the test properly).